src/VisualStudio/Core/Def/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -911,10 +915,70 @@ protected override void AddExistingDocument(DocumentId documentId, string filePa
                 throw new Exception(ServicesVSResources.Could_not_find_location_of_folder_on_disk);
             }
 
-            return AddDocumentToProjectItems(folder.ProjectItems, documentId, folderPath, documentName, initialText, filePath, documentKind);
+            AddDocumentToProjectItems(folder.ProjectItems, documentId, folderPath, documentName, initialText, filePath, documentKind);
+        }
+
+        private bool TryAddEditorConfigToSolutionItems(
+            DocumentInfo documentInfo,
+            SourceText text)
+        {
+            // We're going to see if this is an .editorconfig being added at the solution level, and if so process it specially; so verify first that it is one
+            if (CurrentSolution.FilePath is null || documentInfo.FilePath is null)
+                return false;
+
+            if (PathUtilities.GetFileName(documentInfo.FilePath) != ".editorconfig")
+                return false;
+
+            if (IOUtilities.PerformIO(() => File.Exists(documentInfo.FilePath)))
+                return false;
+
+            var solutionDirectory = PathUtilities.GetDirectoryName(this.CurrentSolution.FilePath);
+            if (solutionDirectory != PathUtilities.GetDirectoryName(documentInfo.FilePath))
+                return false;
+
+            // Double check too that this isn't a case of the .csproj being in the same folder of the .sln, at which point it's reasonable
+            // just to add this to the project file.
+            if (PathUtilities.GetDirectoryName(CurrentSolution.GetProject(documentInfo.Id.ProjectId)?.FilePath) == solutionDirectory)
+                return false;
+
+            // All checks pass, so let's treat this special.
+            var dte = _threadingContext.JoinableTaskFactory.Run(() => _asyncServiceProvider.GetServiceAsync<SDTE, EnvDTE.DTE>(_threadingContext.JoinableTaskFactory));
+
+            const string SolutionItemsFolderName = "Solution Items";
+
+            var projects = dte.Solution.Projects.OfType<EnvDTE.Project>();
+            var solutionItemsFolder = projects.FirstOrDefault(static p => p.Kind == EnvDTE.Constants.vsProjectKindSolutionItems && p.Name == SolutionItemsFolderName);
+
+            if (solutionItemsFolder != null)
+            {
+                foreach (ProjectItem projectItem in solutionItemsFolder.ProjectItems)
+                {
+                    if (projectItem.Name == ".editorconfig")
+                    {
+                        // It's already added, don't need to do anything further.
+                        return true;
+                    }
+                }
+            }
+            else
+            {
+                solutionItemsFolder = ((EnvDTE80.Solution2)dte.Solution).AddSolutionFolder(SolutionItemsFolderName);
+            }
+
+            AddDocumentToProjectItems(
+                solutionItemsFolder.ProjectItems,
+                documentInfo.Id,
+                folderPath: null,
+                ".editorconfig",
```

Comment:
constant for this string that you've used in several places.

Reply:
Switched this out for documentInfo.Name instead.